src/Components/Aspire.Azure.Messaging.EventHubs/EventProcessorClientComponent.cs

```diff
@@ -0,0 +1,95 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Azure.Messaging.EventHubs;
+using Azure.Core;
+using Azure.Core.Extensions;
+using Azure.Messaging.EventHubs;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Extensions.Hosting;
+
+internal sealed class EventProcessorClientComponent(IConfiguration builderConfiguration)
+    : EventHubsComponent<EventProcessorClient, EventProcessorClientOptions>
+{
+    // cannot be in base class as source generator chokes on generic placeholders
+    protected override void BindClientOptionsToConfiguration(IAzureClientBuilder<EventProcessorClient, EventProcessorClientOptions> clientBuilder, IConfiguration configuration)
+    {
+#pragma warning disable IDE0200 // Remove unnecessary lambda expression - needed so the ConfigBinder Source Generator works
+        clientBuilder.ConfigureOptions(options => configuration.Bind(options));
+#pragma warning restore IDE0200
+    }
+
+    protected override IAzureClientBuilder<EventProcessorClient, EventProcessorClientOptions> AddClient<TBuilder>(TBuilder azureFactoryBuilder, AzureMessagingEventHubsSettings settings,
+        string connectionName, string configurationSectionName)
+    {
+        return azureFactoryBuilder.RegisterClientFactory<EventProcessorClient, EventProcessorClientOptions>(
+            (options, cred) =>
+            {
+                var connectionString = settings.ConnectionString;
+                if (string.IsNullOrEmpty(connectionString) && string.IsNullOrEmpty(settings.Namespace))
+                {
+                    throw new InvalidOperationException(
+                        $"A EventHubProducerClient could not be configured. Ensure valid connection information was provided in 'ConnectionStrings:{connectionName}' or specify a 'ConnectionString' or 'Namespace' in the '{configurationSectionName}' configuration section.");
+                }
+
+                if (string.IsNullOrEmpty(settings.EventHubName))
+                {
+                    throw new InvalidOperationException(
+                        $"A EventProcessorClient could not be configured. Ensure a valid EventHubName was provided in the '{configurationSectionName}' configuration section.");
+                }
+
+                // configure processor identifier
+                options.Identifier ??= $"{settings.EventHubName}-{settings.ConsumerGroup ?? "default"}";
```

Comment:
This is a nice touch, though we'll want something that can be unique otherwise we'll see problems with load balancing and ownership.   Maybe stick the machine name and a random slug in here too?   

That'll help keep a meaningful name for logs, but also ensure that we don't see collisions for processors running in the same cluster.

Reply:
This look okay to you?

     var slug = Guid.NewGuid().ToString().Substring(24);
     options.Identifier ??= $"{Environment.MachineName}-{settings.EventHubName}-{settings.ConsumerGroup ?? "default"}-{slug}";

Reply:
Yup, that looks good!